src/sd.jl

```diff
@@ -25,3 +25,113 @@ function addconstraint(m::Model, c::SDVariableConstraint)
     cref = MOI.addconstraint!(m.instance, MOI.VectorOfVariables([instancereference(c.Q[i, j]) for j in 1:n for i in 1:j]), MOI.PositiveSemidefiniteConeTriangle(n))
     return ConstraintRef(m, cref)
 end
+
+##########################################################################
+# SDConstraint is a (dual) semidefinite constraint of the form
+# ∑ cᵢ Xᵢ ⪰ D, where D is a n×n data matrix, cᵢ are scalars,
+# and Xᵢ are n×n symmetric variable matrices. The inequality
+# is taken w.r.t. the psd partial order.
+struct SDConstraint{T<:AbstractJuMPScalar, MT<:AbstractMatrix{T}} <: AbstractConstraint
```

Comment:
Can we just reuse `VectorAffExprConstraint` here?

Reply:
Not really since here we have a matrix of `AffExpr` and not a vector of `AffExpr`. Moreover `VectorAffExprConstraint` assumes that the vector will be passed as is to the instance while here we need to only take the upper triangular part and check symmetry.